### PR TITLE
[Quit Test Flow] Completed the Quit Test Flow

### PR DESCRIPTION
**Pull request by fnormand01**
# Description

I completed the quit test flow by adding the generic popup box component when selecting the **Quit test** button. This popup box also contains an error/danger message, so I also used the generic system message component to get that.

## Type of change
- New feature (non-breaking change which adds functionality)
- Code cleanliness or refactor

## Screenshot

**English Version:**

image

**French Version:**

image

**When less than all the checkboxes are selected (the button still disabled):**

image

**When all checkboxes are selected (the button becomes enabled):**

image

***Note that the translation hasn't been done in this PR, except for the buttons. This translation will be done in another PR.**

# Testing

Manual steps to reproduce this functionality:

1.  Start the eMIB sample test.
2.  Complete the pre-test process.
3.  Click **Quit test** and make sure that all buttons / checkboxes in the popup box work well.
4.  Also, make sure that the checkbox states are resetting when closing the popup box (check at least one checkbox before closing the popup box to test that)

Screenshot of unit tests run passing:

image

# Checklist

Applicable for all code changes.

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~~I have made corresponding changes to the documentation~~
- [x] My changes generate no new compiler warnings
- [ ] ~~My changes generate no new accessibility errors and/or warnings~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have researched WCAG2.1 accessibility standards and met them in this PR (can be navigated using a keyboard)
- [x] My changes look good on IE 10+, Firefox, and Chrome

### CODE REVIEW

File: frontend/src/components/eMIB/Emib.jsx [hunk 1]
```diff
@@ -53,12 +78,84 @@ class Emib extends Component {
...
 
-  openPopup = () => {
-    this.setState({ showPopupBox: true });
+  openSubmitPopup = () => {
+    this.setState({ showSubmitPopup: true });
   };
 
   closePopup = () => {
-    this.setState({ showPopupBox: false });
+    this.setState({ showSubmitPopup: false });
```
Comment (caleybrock) [inline frontend/src/components/eMIB/Emib.jsx]: I think there's actually no need to have multiple setState's in a row. You should be able to do this here and below:
```
this.setState({
  showSubmitPopup: false,
  showQuitPopup: fase
});
```

Comment (fnormand01) [reply frontend/src/components/eMIB/Emib.jsx]: Simplified all the multiple **setState**.
See commit #https://github.com/code-for-canada/project-thundercat/pull/129/commits/350483793ba562d8e94259fb53298e69e55e60bf

File: frontend/src/components/eMIB/Emib.jsx [hunk 2]
```diff
@@ -53,12 +78,94 @@ class Emib extends Component {
...
+    this.setState({ showSubmitPopup: false, showQuitPopup: false });
+    //reset all checkbox states on close
+    this.resetCheckboxStates();
+  };
+
+  openQuitPopup = () => {
+    this.setState({ showQuitPopup: true });
+  };
+
+  firstCheckboxToggle = () => {
```
Comment (caleybrock) [inline frontend/src/components/eMIB/Emib.jsx]: I think the logic makes sense here, but there's quite a bit of duplicated code. Let's pair on this and see if we can come up with something a little lighter?

Comment (fnormand01) [reply frontend/src/components/eMIB/Emib.jsx]: We did some pair programming to simplified that. See last 6 commits.

**Comment by fnormand01** [comment]
All tests still passe after the refactoring updates:

image

**Comment by fnormand01** [comment]
@caleybrock @MichaelCherry, could you please take another quick look to make sure everything works well, since we did a lot of changes?

**Comment by fnormand01** [comment]
Known Bug: When you click on **Français** and then click on **Quitter le test**, you'll see that the text of the checkboxes is not translated. However, if you close this popup box and reopen it, you'll see that the text will be updated, meaning in French.

Same behavior with French to English action.

This bug will be fix in another PR.

**Pull request merged**